src/background/quality/providers/QualityProvider.ts

```diff
@@ -1 +1,60 @@
-export class QualityProvider {}
+import { Storage } from "webextension-polyfill-ts";
+
+import { Quality } from "../../../types/Quality";
+import { StorageError } from "../../../types/errors/StorageError";
+
+export class QualityProvider {
+  private isCacheActive = true;
+
+  constructor(private storageArea: Storage.StorageArea) {}
+
+  updateIsCacheActive(isCacheActive: boolean): void {
+    this.isCacheActive = isCacheActive;

```

Comment:
Would the user expect the cached domains to be deleted upon deactivation?

Reply:
I don't think so, since there is an extra button in the same view to exactly do that. There is nothing implying that this should be the case.